### PR TITLE
Add Recaf format support

### PR DESCRIPTION
**Pull request by Toshimichi0915**

### CODE REVIEW

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingFormat.java [hunk 1]
```diff
@@ -37,11 +37,11 @@ public enum MappingFormat {
 		this.reader = reader;
 	}
 
-	public void write(EntryTree<EntryMapping> mappings, Path path, ProgressListener progressListener, MappingSaveParameters saveParameters)  {
+	public void write(EntryTree<EntryMapping> mappings, Path path, ProgressListener progressListener, MappingSaveParameters saveParameters) {
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingFormat.java:40]: Undo these whitespace changes.

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/recaf/RecafMappingsReader.java [hunk 2]
```diff
@@ -22,40 +22,40 @@
...
-            }
-
-            Matcher classMatcher = CLASS_PATTERN.matcher(line);
-            if (classMatcher.find()) {
-                mappings.insert(new ClassEntry(classMatcher.group(1)), new EntryMapping(classMatcher.group(2)));
-            }
-        }
-        return mappings;
-    }
+	public static final RecafMappingsReader INSTANCE = new RecafMappingsReader();
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/recaf/RecafMappingsReader.java:25]: Undo these whitespace changes, at somepoint we should add checkstyle to enfoce a consitent format, but not now.

**Review by modmuss50 — APPROVED** [review]
LGTM other than the whitespaces changes to those two files.

Might be nice to get Recaf support into mapping-io as well 👍

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/recaf/RecafMappingsReader.java (hunk 2 above)
Comment (Toshimichi0915) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/recaf/RecafMappingsReader.java:25]: Should I use space for all new files?

File: enigma/src/main/resources/lang/zh_cn.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
 	"mapping_format.tiny_file": "Tiny 文件",
 	"mapping_format.srg_file": "SRG 文件",
 	"mapping_format.proguard": "Proguard 文件",
+	"mapping_format.recaf": "Recaf",
```
Comment (2xsaiko) [inline enigma/src/main/resources/lang/zh_cn.json:11]:
```suggestion
	"mapping_format.recaf": "Recaf 文件",
```
The others say 文件 too, to keep it consistent?

**Comment by 2xsaiko** [comment]
Thanks!

File: enigma/build.gradle [hunk 4]
```diff
@@ -12,6 +12,8 @@ dependencies {
     implementation 'net.fabricmc:cfr:0.1.0'
 
     proGuard 'com.guardsquare:proguard-base:7.2.0-beta2'
+
+    testImplementation 'com.google.jimfs:jimfs:1.2'
```
Comment (Col-E) [inline enigma/build.gradle:16]: Using JIMFS just to read from `test/resources/recaf.mappings` is over-complicating a simple IO task.

**Comment by Toshimichi0915** [comment]
I understand your thoughts. However, I thought the name "Simple" was kinda confusing to some users and as far as I! know that mapping format is used only in Recaf.

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingFormat.java [hunk 5]
```diff
@@ -24,7 +25,8 @@ public enum MappingFormat {
 	TINY_V2(new TinyV2Writer("intermediary", "named"), new TinyV2Reader()),
 	TINY_FILE(TinyMappingsWriter.INSTANCE, TinyMappingsReader.INSTANCE),
 	SRG_FILE(SrgMappingsWriter.INSTANCE, null),
-	PROGUARD(null, ProguardMappingsReader.INSTANCE);
+	PROGUARD(null, ProguardMappingsReader.INSTANCE),
+	RECAF(null, RecafMappingsReader.INSTANCE);
```
Comment (NebelNidas) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingFormat.java:29]: You've forgotten to add your RecafMappingsWriter here

**Pull request merged**